javascript/283-Move-Zeroes.js

```diff
@@ -1,3 +1,23 @@
+/**
+ * Linear Time
+ * Time Complexity  O(N) | Space Complexity O(N);
+ * https://leetcode.com/problems/move-zeroes/
+ * @param {number[]} nums
+ * @return {void} Do not return anything, modify nums in-place instead.
+ */
+var moveZeroes = function(nums) {
+
+    const zeroAtTheEnd = Array(nums.length).fill(0);
+    let left = 0;
+    for (let i = 0; i < nums.length; i++) {
```

Comment:
Most of the solutions for 2 pointer and cyclic sort are in while loop. Easier to remember that pattern over for loop

Reply:
Look at the Space O(1) if condition

Reply:
Hey @aakhtar3, check now. Is this ok?

Reply:
Let's do this
```js
const arr = new Array(nums.length).fill(0);

let [ left, right ] = [ 0, 0 ];

while (right < nums.length) {
	const isZero = (nums[right] === 0);
	if (!isZero) {
		arr[left] = nums[right];
		left++;
	}

	right++;
}

return arr;
```

Reply:
That could work, let me try to submit it on leetCode.

Reply:
Oh, shoot! we won't be able to because we're creating a new array.

Reply:
But I did try some test cases and it seems to run just fine. I'll update the code.

Reply:
Please check now @aakhtar3